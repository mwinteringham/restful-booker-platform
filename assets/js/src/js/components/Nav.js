import React from 'react';
import Cookies from 'universal-cookie';
import { Link } from 'react-router-dom';
import { API } from '../libs/Api';
import Notification from './Notification.js';

const Nav = ({setAuthenticate, isAuthenticated, setCount, count}) => {
	
	const doLogout = () => {
		const cookies = new Cookies();
		let token = cookies.get('token')

		API.postLogout(setAuthenticate, token);
	}

	return(
			<nav className="navbar navbar-expand-md navbar-dark bg-dark">
				<div className="navbar-collapse collapse w-100 order-1 order-md-0 dual-collapse2">
					{isAuthenticated && (
						<ul className="navbar-nav mr-auto">
								<li className="nav-item">
									<Link className="nav-link" to="/admin/">Rooms</Link>
								</li>
								<li className="nav-item">
									<Link className="nav-link" to="/admin/report" id="reportLink">Report</Link>
								</li>
								<li className="nav-item">
									<Link className="nav-link" to="/admin/branding" id="brandingLink">Branding</Link>
								</li>
						</ul>
					)}
				</div>
				<div className="mx-auto order-0">
						<a className="navbar-brand mx-auto" href="#">B&B Booking Management</a>
						<button className="navbar-toggler" type="button" data-toggle="collapse" data-target=".dual-collapse2">
								<span className="navbar-toggler-icon"></span>
						</button>
				</div>
				<div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
					{isAuthenticated && (
						<ul className="navbar-nav ml-auto">
								<li className="nav-item">
									<Notification setCount={setCount} count={count} />
								</li>
								<li className="nav-item">
										<a className="nav-link" id="frontPageLink" href="/">Front Page</a>
								</li>
								<li className="nav-item">
										<a className="nav-link" href="#/admin" onClick={doLogout}>Logout</a>
								</li>
						</ul>
					)}
				</div>
		</nav>
	);
}

export default Nav;
