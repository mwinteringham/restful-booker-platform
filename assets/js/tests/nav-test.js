import React from 'react';
import Nav from '../src/js/components/Nav.js';
import { BrowserRouter } from 'react-router-dom';
import {
    render
  } from '@testing-library/react'

test('Nav bar renders', () => {
    const {asFragment} = render(
        <BrowserRouter><Nav setCount={() => {}} isAuthenticated={true} /></BrowserRouter>
    )

    expect(asFragment()).toMatchSnapshot();
})