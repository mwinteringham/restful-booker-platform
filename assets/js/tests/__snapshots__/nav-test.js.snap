// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar renders 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      class="navbar-collapse collapse w-100 order-1 order-md-0 dual-collapse2"
    >
      <ul
        class="navbar-nav mr-auto"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/admin/"
          >
            Rooms
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/admin/report"
            id="reportLink"
          >
            Report
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/admin/branding"
            id="brandingLink"
          >
            Branding
          </a>
        </li>
      </ul>
    </div>
    <div
      class="mx-auto order-0"
    >
      <a
        class="navbar-brand mx-auto"
        href="#"
      >
        B&B Booking Management
      </a>
      <button
        class="navbar-toggler"
        data-target=".dual-collapse2"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
    </div>
    <div
      class="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/admin/messages"
          >
            <i
              class="fa fa-inbox"
              style="font-size: 1.5rem;"
            />
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/"
            id="frontPageLink"
          >
            Front Page
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/admin"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
